tests/unit/repo/test_repo.py

```diff
@@ -82,3 +82,18 @@ def test_collect_optimization(tmp_dir, dvc, mocker):
     # Should read stage directly instead of collecting the whole graph
     dvc.collect(stage.path)
     dvc.collect_granular(stage.path)
+
+
+def test_skip_graph_checks(tmp_dir, dvc, mocker, run_copy):
+    (stage,) = tmp_dir.dvc_gen("foo", "foo text")
+
+    # Error out on graph collection and raise a skip graph checks flag
+    mocker.patch(
+        "dvc.repo.Repo._collect_graph",
+        side_effect=Exception("Should not collect"),
+    )
```

Comment:
It would be better to just assert that this wasn't called rather than putting this exception here, as we might have some logic catching this in our code (yes, we don't do that now and catching Exception is a bad thing, but still), which would result in a faulty test. Plus it would be more explicit to use `assert not m.called` rather than this unnecessary hack.